lib/ui/src/components/sidebar/Search.tsx

```diff
@@ -324,10 +324,12 @@ export const Search = React.memo<{
           getRootProps,
           highlightedIndex,
         }) => {
+          const isBrowsing = !isOpen && document.activeElement !== inputRef.current;
           const filter = inputValue ? inputValue.trim() : '';
           let results: DownshiftItem[] = filter ? getResults(filter) : [];
 
-          syncUrlToFilter(filter);
+          syncUrlToFilter(isBrowsing ? null : filter);
+          api.setQueryParams({ filter: isBrowsing ? null : filter });
```

Comment:
I think we don't even need `syncUrlToFilter` and can just rely on `api.setQueryParams` to do the replaceState for us. The only thing is the value cannot be `null` because `setQueryParams` will ignore those. Probably you can just set it to `undefined`.